### PR TITLE
Consider port differences between versions when getting log

### PR DESCRIPTION
**Pull request by hmusum**

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java [hunk 1]
```diff
@@ -1169,33 +1177,40 @@ private void logConfigChangeActions(ConfigChangeActions actions, DeployLogger lo
...
-        int port = servicePort(logService);
-        return HttpURL.create(Scheme.http, DomainName.of(logServerHostInfo.getHostname()), port, HttpURL.Path.parse("logs"));
+                .map(hostInfo -> hostInfo.getServices().stream()
+                        .filter(service -> LOGSERVER_CONTAINER.serviceName.equals(service.getServiceType()))
+                        .findFirst()
+                        .or(() -> hostInfo.getServices().stream()
+                                .filter(service -> CONTAINER.serviceName.equals(service.getServiceType()))
+                                .findFirst())
+                        .orElseThrow(() -> new IllegalArgumentException("No container running on logserver host")))
+                .collect(Collectors.toMap(ServiceInfo::getHostName, this::servicePort, (a, b) -> a));
```
Comment (jonmv) [inline configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java:1213]: I think you actually need `groupingBy` here?

Comment (jonmv) [reply configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java:1213]: But then it's simpler to just return a list of pairs.

Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java]: Right, there _could_ be different hostnames, I guess. Will fix

**Comment by hmusum** [comment]
@jonmv please take a look again

**Review by jonmv — APPROVED** [review]
Nice!

**Pull request merged**